plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveTransaction.java

```diff
@@ -63,13 +65,24 @@ public AcidTransaction getTransaction()
 
     public ValidTxnWriteIdList getValidWriteIds(HiveMetastoreClosure metastore, HiveTableHandle tableHandle)
     {
+        boolean partitioned = false;
+        if (tableHandle.getPartitions().isPresent()) {
+            if (tableHandle.getPartitions().get().stream().anyMatch(partition -> partition.getPartitionId().equals(UNPARTITIONED_ID))) {
+                // if there is just partition with `UNPARTITIONED_ID` id in table handle treat table as unpartitioned
+                checkArgument(tableHandle.getPartitions().get().size() == 1, "expected only one hive partition");
+            }
+            else {
+                partitioned = true;
+            }
+        }
```

Comment:
it looks as if `partitioned` could remain false for partitioned table if `tableHandle.getPartitions()` is not filled in yet.

- checking `io.trino.plugin.hive.HiveTableHandle.getPartitionColumns().isEmpty()` seems like the more direct way to determine whether table is partitioned
- can `tableHandle.getPartitions()` be not filled yet at this point? if so, then


```
List<..> lockedTables;
List<..> lockedPartitions;
boolean partitioned = tableHandle.getPartitionColumns().isEmpty();
if (partitioned && tableHandle.getPartitions().isPresent()) {
 . ...
} 
else {
 ..
}
```

As a follow up, it would be nice to reconsider whether we want to fill `tableHandle.getPartitions()` for unpartitioned tables.

Reply:
Nicer. Thanks.